### PR TITLE
nbdkit: init at 1.40.4

### LINKED ISSUE #272120: Package request: nbdkit
**Project description**

nbdkit is an NBD server.  NBD — Network Block Device — is a protocol
for accessing Block Devices (hard disks and disk-like things) over a
Network.
The key features of nbdkit are:

- Multithreaded NBD server written in C with good performance.
- Minimal dependencies for the basic server.
- Liberal license (BSD) allows nbdkit to be linked to proprietary
- libraries or included in proprietary code.
- Well-documented, simple plugin API with a stable ABI guarantee.
- Lets you export “unconventional” block devices easily.
- You can write plugins in C/C++, Go, Lua, Perl, Python, OCaml, Ruby,
- Rust, shell script or Tcl.
- Filters can be stacked in front of plugins to transform the output.

**Metadata**

* homepage URL: https://gitlab.com/nbdkit/nbdkit
* source URL: https://gitlab.com/nbdkit/nbdkit
* license: Red Hat License (not sure what it is) https://gitlab.com/nbdkit/nbdkit/-/blob/32a9ee6650654469cd591a3ae26842c54f898392/LICENSE
* platforms: unix, linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by lukts30**
Closes #272120
Also needed for virt-v2v.
For now without any bindings.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lukts30** [comment]
ofborg fails because I added myself in https://github.com/NixOS/nixpkgs/pull/345977 to [maintainer-list.nix](https://github.com/NixOS/nixpkgs/pull/345977/commits/c14eb151f075478a41d7500eb46b5fa3b82153c6) which in not merged yet.

File: pkgs/by-name/nb/nbdkit/package.nix [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  pname = "nbdkit";
+  version = "1.40.4";
+
+  src = fetchurl {
+    url = "https://download.libguestfs.org/nbdkit/${lib.versions.majorMinor finalAttrs.version}-stable/nbdkit-${finalAttrs.version}.tar.gz";
+    hash = "sha256-hGoc34F7eAvHjdQHxcquNJhpwpL5CLfv2DBZKVmpcpw=";
+  };
+
+  prePatch = ''
+    # some scripts hardcore /usr/bin/env which is not available in the build env
```
Comment (bjornfor) [inline pkgs/by-name/nb/nbdkit/package.nix:31]: s/hardcore/hardcode/ :smile:

File: pkgs/by-name/nb/nbdkit/package.nix [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    "--disable-example4"
+    "--disable-floppy"
+  ];
+
+  installFlags = [ "bashcompdir=$(out)/share/bash-completion/completions" ];
+
+  postInstall = ''
+    for bin in $out/bin/*; do
+      wrapProgram "$bin" \
+        --prefix PATH     : "$out/bin:${e2fsprogs}/bin:${cdrkit}/bin"
```
Comment (bjornfor) [inline pkgs/by-name/nb/nbdkit/package.nix:83]: * Shell indent is typically 4 chars?
* Why lots of space in "--prefix PATH [spaaaace] :"?
* Use makeBinPath? `--prefix PATH : "$out/bin:${lib.makeBinPath [ e2fsprogs cdrkit ]}"`

File: pkgs/by-name/nb/nbdkit/package.nix [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    done
+  '';
+
+  passthru.tests.version = testers.testVersion { package = finalAttrs.finalPackage; };
+
+  meta = {
+    homepage = "https://gitlab.com/nbdkit/nbdkit";
+    description = "NBD server with stable plugin ABI and permissive license.";
+    license = with lib.licenses; bsd3;
+    # maintainers = with maintainers; [ lukts30 ];
```
Comment (bjornfor) [inline pkgs/by-name/nb/nbdkit/package.nix:93]: You can rebase onto master now, and enable this maintainers line.

File: pkgs/by-name/nb/nbdkit/package.nix [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+          ]
+        }"
+    done
+  '';
+
+  passthru.tests.version = testers.testVersion { package = finalAttrs.finalPackage; };
+
+  meta = {
+    homepage = "https://gitlab.com/nbdkit/nbdkit";
+    description = "NBD server with stable plugin ABI and permissive license.";
```
Comment (bjornfor) [inline pkgs/by-name/nb/nbdkit/package.nix:94]: Please drop dot at the end of description. See https://ryantm.github.io/nixpkgs/stdenv/meta/#var-meta-description.

**Review by bjornfor — APPROVED** [review]

**Pull request merged**